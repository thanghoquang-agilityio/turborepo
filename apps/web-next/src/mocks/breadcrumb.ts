export const MOCK_BREADCRUMB_ITEMS = [
  { name: 'Products', link: 'products' },
  { name: 'Detail Product', link: '7' },
]
