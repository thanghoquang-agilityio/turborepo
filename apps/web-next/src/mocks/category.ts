export const MOCK_CATEGORY_LIST = [
  {
    name: 'Kid',
    value: 'kid',
    total: 18,
  },
  {
    name: 'Man',
    value: 'man',
    total: 12,
  },
  {
    name: 'Woman',
    value: 'woman',
    total: 23,
  },
  {
    name: 'Sport',
    value: 'sport',
    total: 34,
  },
]

export const MOCK_CATEGORY_RESPONSE = [
  {
    name: 'Kid',
    value: 'kid',
    products: {
      data: [],
    },
  },
  {
    name: 'Man',
    value: 'man',
    products: {
      data: [],
    },
  },
  {
    name: 'Woman',
    value: 'woman',
    products: {
      data: [],
    },
  },
  {
    name: 'Sport',
    value: 'sport',
    products: {
      data: [],
    },
  },
]
