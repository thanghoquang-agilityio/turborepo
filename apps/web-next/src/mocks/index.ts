export * from './breadcrumb'
export * from './cart'
export * from './category'
export * from './image'
export * from './product'
export * from './radio-group'
export * from './user'
