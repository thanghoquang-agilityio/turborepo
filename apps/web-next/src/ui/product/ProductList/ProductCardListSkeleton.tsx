import { Skeleton } from '@nextui-org/react'

import { PAGE_SIZE_DEFAULT } from '@/constants'
// Styles
import '@/styles/card-list.css'
import '@/styles/card.css'

export const ProductCardListSkeleton = ({
  additionalClass,
}: {
  additionalClass?: string
}) => {
  return (
    <div
      className={`card-list mx-auto mb-[110px] grid w-fit grid-cols-2 gap-x-[20px] gap-y-[30px] ${additionalClass}`}
    >
      {Array(PAGE_SIZE_DEFAULT)
        .fill(0)
        .map((_, index) => (
          <div
            key={`card-product-skeleton-${index}`}
            className={`card flex max-w-[305px] flex-col items-center overflow-hidden rounded-large border-medium border-background-300`}
          >
            <Skeleton className='card-image bg-default-200 h-[185px] w-full md:h-[303px] md:w-[305px]' />
            <div className='w-full p-5'>
              <div className='h-12 w-full space-y-[5px]'>
                <Skeleton className='bg-default-200 h-[18px] w-full rounded-large' />
                <Skeleton className='bg-default-200 h-[16px] w-[150px] rounded-large' />
              </div>
              <Skeleton className='bg-default-200 h-5 w-[100px] rounded-large' />
            </div>
            <div className='card-content mb-6 mt-1 flex w-full flex-col items-end gap-3 px-5 md:flex-row md:justify-between'>
              <Skeleton className='bg-default-200 h-9 w-full max-w-full rounded-large md:max-w-[128px]' />
              <Skeleton className='bg-default-200 h-9 w-full max-w-full rounded-large md:max-w-[128px]' />
            </div>
          </div>
        ))}
    </div>
  )
}
