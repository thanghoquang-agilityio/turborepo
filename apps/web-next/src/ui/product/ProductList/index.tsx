'use client'

import { Suspense, useCallback, useMemo, useTransition } from 'react'

import dynamic from 'next/dynamic'
import { usePathname, useRouter, useSearchParams } from 'next/navigation'

// Components
import { Skeleton } from '@nextui-org/react'

import { Button, Content, Text } from '@/components'
import {
  COMMENTS_DEFAULT,
  PAGE_DEFAULT,
  PAGE_SIZE_DEFAULT,
  STAR_DEFAULT,
} from '@/constants'
// Styles
import '@/styles/card-list.css'
// Types
import { FilterItem, MetaResponse, ProductModel } from '@/types'

import { ProductCardListSkeleton } from './ProductCardListSkeleton'

const FilterList = dynamic(() => import('@/components/FilterList'))
const ProductCard = dynamic(() => import('../ProductCard'))
const Pagination = dynamic(() => import('@/components/Pagination'))
const SelectMultiple = dynamic(
  () => import('@/components/common/SelectMultiple')
)

interface ProductListProps extends MetaResponse {
  products: ProductModel[]
  categories: FilterItem[]
  valueCategories?: string[]
  isFilter?: boolean
}

const ProductList = ({
  products,
  pagination,
  categories,
  isFilter,
  valueCategories = [],
}: ProductListProps) => {
  const {
    page = PAGE_DEFAULT,
    pageSize = PAGE_SIZE_DEFAULT,
    total = PAGE_SIZE_DEFAULT * PAGE_DEFAULT,
  } = pagination

  const [isPending, startTransition] = useTransition()

  const searchParams = useSearchParams() ?? ''
  const pathname = usePathname() ?? ''
  const router = useRouter()

  const params = useMemo(
    () => new URLSearchParams(searchParams),
    [searchParams]
  )

  const handleReplaceURL = useCallback(
    (params: URLSearchParams) => {
      startTransition(() => {
        router.replace(`${pathname}?${params.toString()}`)
      })
    },
    [pathname, router]
  )

  const handlePageChange = useCallback(
    (page: number) => {
      if (page === 1) {
        params.delete('page')
      } else if (!params.getAll('page').includes(String(page))) {
        params.append('page', String(page))
      }

      handleReplaceURL(params)
    },
    [handleReplaceURL, params]
  )

  const handleClearFilter = useCallback(() => {
    startTransition(() => {
      router.replace(`${pathname}`)
    })
  }, [pathname, router])

  const classProductList = isFilter
    ? '2xl:grid-cols-3 lg:grid-cols-2'
    : '2xl:grid-cols-4 lg:grid-cols-3'

  const classLayoutProductList = isFilter
    ? 'flex lg:flex-row flex-col lg:gap-[21px] lg:items-start items-center'
    : ''

  return (
    <div
      className={`container w-full max-w-full py-8 lg:py-12 xl:py-16 ${classLayoutProductList}`}
    >
      {isFilter && (
        <div className='mb-8 flex w-[304px] flex-col'>
          <div className='flex justify-between'>
            <Text
              variant='default'
              className='text-xl font-medium text-secondary-200'
            >
              Filters
            </Text>

            <Button
              variant='bordered'
              size='sm'
              additionalClass='min-w-[95px] w-[95px] text-foreground-100 font-normal'
              onClick={handleClearFilter}
            >
              Clear All
            </Button>
          </div>
          <div className='mt-5 w-[304px] rounded-medium border-medium border-background-300 p-4'>
            <Suspense>
              <FilterList
                title='Category'
                isOpen={true}
                items={categories}
                defaultValue={valueCategories}
                handleReplaceURL={handleReplaceURL}
              />
            </Suspense>
          </div>
        </div>
      )}

      <div className='w-full'>
        <div
          className={`flex flex-col items-center gap-4 sm:flex-row lg:justify-between ${isFilter ? 'mb-[20px] justify-between' : 'mb-6 justify-end xl:mb-12'}`}
        >
          <div className='flex gap-3'>
            {isFilter ? (
              <>
                {isPending ? (
                  <Skeleton className='bg-default-200 hidden h-[37px] w-[200px] rounded-large sm:block' />
                ) : (
                  <Text variant='title' size='sm'>
                    {page === 1
                      ? `Showing ${total < pageSize ? total : pageSize} Result from total ${total}`
                      : `Showing ${pageSize * (page - 1) + 1} to ${pageSize * page > total ? total : pageSize * page} Result from total ${total}`}
                  </Text>
                )}
              </>
            ) : (
              <Suspense>
                <SelectMultiple
                  placeholder='Category'
                  items={categories}
                  additionalClass='w-[160px] block'
                  defaultValue={valueCategories}
                  handleReplaceURL={handleReplaceURL}
                />
                <SelectMultiple
                  placeholder='Price'
                  items={[]}
                  additionalClass='w-[88px] lg:block hidden'
                  isDisabled={true}
                />
                <SelectMultiple
                  placeholder='Review'
                  items={[]}
                  additionalClass='w-[102px] lg:block hidden'
                  isDisabled={true}
                />
                <SelectMultiple
                  placeholder='Color'
                  items={[]}
                  additionalClass='w-[98px] lg:block hidden'
                  isDisabled={true}
                />
                <SelectMultiple
                  placeholder='Material'
                  items={[]}
                  additionalClass='w-[108px] lg:block hidden'
                  isDisabled={true}
                />
                <SelectMultiple
                  placeholder='Offer'
                  items={[]}
                  additionalClass='w-[88px] lg:block hidden'
                  isDisabled={true}
                />
              </Suspense>
            )}
          </div>
          <Suspense>
            <SelectMultiple
              placeholder='Headphone type'
              items={[]}
              additionalClass={`w-[166px] bg-background-100 border-small border-foreground-300 ${isFilter ? 'sm:block hidden' : 'lg:block hidden'}`}
              isDisabled={true}
            />
          </Suspense>
        </div>
        {isPending ? (
          <ProductCardListSkeleton additionalClass={classProductList} />
        ) : (
          <>
            {products.length > 0 ? (
              <>
                <div
                  className={`card-list mx-auto mb-[110px] grid w-fit grid-cols-2 gap-x-[20px] gap-y-[30px] ${classProductList}`}
                >
                  {products.map((product: ProductModel) => {
                    const {
                      id = '',
                      name = '',
                      price = 0,
                      star = STAR_DEFAULT,
                      comments = COMMENTS_DEFAULT,
                      image
                    } = product

                    const imageUrl = image?.url
                    
                    const variantsCount = product.productVariants?.length || 0

                    return (
                      <Suspense key={`card-product-suspense-${id}`}>
                        <ProductCard
                          key={`card-product-${id}`}
                          id={id.toString()}
                          image={imageUrl}
                          name={name}
                          price={price}
                          star={star}
                          comments={comments}
                          variants={variantsCount}
                        />
                      </Suspense>
                    )
                  })}
                </div>
                <Suspense>
                  <Pagination
                    currentPage={page}
                    pageSize={pageSize}
                    totalCount={total}
                    onPageChange={handlePageChange}
                  />
                </Suspense>
              </>
            ) : (
              <div className='flex h-[320px] min-h-[320px] flex-col items-center justify-center gap-4'>
                <Content title='Empty product' subTitle='' />
              </div>
            )}
          </>
        )}
      </div>
    </div>
  )
}

export default ProductList
