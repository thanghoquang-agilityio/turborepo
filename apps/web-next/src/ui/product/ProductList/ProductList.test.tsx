import { render } from '@testing-library/react'

import { MOCK_CATEGORY_RESPONSE, MOCK_PRODUCTS_RESPONSE } from '@/mocks'

import ProductList from '.'

jest.mock('next/navigation', () => ({
  usePathname: jest.fn().mockImplementation(() => '/products'),
  useRouter: jest.fn().mockReturnValue({
    replace: jest.fn(),
  }),
  useSearchParams: jest.fn().mockReturnValue(new URLSearchParams('page=1')),
}))

describe('ProductList', () => {
  const renderComponent = () =>
    render(
      <ProductList
        products={MOCK_PRODUCTS_RESPONSE}
        categories={MOCK_CATEGORY_RESPONSE}
        pagination={{ page: 1, pageSize: 2, total: 5 }}
      />
    )

  it('should match snapshot value error', () => {
    const { container } = render(
      <ProductList
        products={[]}
        categories={[]}
        pagination={{ page: 1, pageSize: 2, total: 5 }}
      />
    )

    expect(container).toMatchSnapshot()
  })

  it('should match snapshot', () => {
    const { container } = renderComponent()

    expect(container).toMatchSnapshot()
  })
})
