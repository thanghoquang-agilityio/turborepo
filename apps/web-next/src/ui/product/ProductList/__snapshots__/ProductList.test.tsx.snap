// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList should match snapshot 1`] = `
<div>
  <div
    class="container w-full max-w-full xl:py-16 lg:py-12 py-8 "
  >
    <div
      class="w-full"
    >
      <div
        class="flex sm:flex-row flex-col gap-4 items-center lg:justify-between justify-end xl:mb-12 mb-6"
      >
        <div
          class="flex gap-3"
        >
          <div
            class="group flex-col transition-background motion-reduce:transition-none !duration-150 group relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)] rounded-[28px] opacity-100 bg-background-400 text-primary-200 font-medium text-sm w-[160px] block"
            data-filled="true"
            data-slot="base"
          >
            <div
              aria-hidden="true"
              data-a11y-ignore="aria-hidden-focus"
              data-testid="hidden-select-container"
              style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
            >
              <input
                style="font-size: 16px;"
                tabindex="-1"
                type="text"
                value=""
              />
              <label>
                <select
                  multiple=""
                  size="4"
                  tabindex="-1"
                  title=""
                >
                  <option />
                  <option
                    value="Category-0"
                  >
                    Kid
                  </option>
                  <option
                    value="Category-1"
                  >
                    Man
                  </option>
                  <option
                    value="Category-2"
                  >
                    Woman
                  </option>
                  <option
                    value="Category-3"
                  >
                    Sport
                  </option>
                </select>
              </label>
            </div>
            <div
              class="w-full flex flex-col"
              data-slot="mainWrapper"
            >
              <button
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-label="Category"
                aria-labelledby="react-aria-:r1v: react-aria-:r1q:"
                class="relative gap-3 w-full inline-flex flex-row items-center shadow-sm tap-highlight-transparent bg-default-100 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 min-h-9 h-9 px-3.5 rounded-[28px]"
                data-slot="trigger"
                data-testid="Category"
                id="react-aria-:r1q:"
                type="button"
              >
                <div
                  class="inline-flex h-full w-[calc(100%_-_theme(spacing.6))] min-h-4 items-center gap-1.5 box-border"
                  data-slot="innerWrapper"
                >
                  <span
                    class="w-full text-left rtl:text-right truncate group-data-[has-value=true]:text-default-foreground text-primary-200 font-medium text-sm"
                    data-slot="value"
                    id="react-aria-:r1v:"
                  >
                    Category
                  </span>
                </div>
                <svg
                  aria-hidden="true"
                  class="absolute right-3 rtl:left-3 rtl:right-[unset] w-4 h-4 transition-transform duration-150 ease motion-reduce:transition-none data-[open=true]:rotate-180"
                  data-slot="selectorIcon"
                  fill="none"
                  focusable="false"
                  height="1em"
                  role="presentation"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="1em"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="group flex-col pointer-events-none transition-background motion-reduce:transition-none !duration-150 group relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)] rounded-[28px] opacity-100 bg-background-400 text-primary-200 font-medium text-sm w-[88px] lg:block hidden"
            data-filled="true"
            data-slot="base"
          >
            <div
              aria-hidden="true"
              data-a11y-ignore="aria-hidden-focus"
              data-testid="hidden-select-container"
              style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
            >
              <input
                disabled=""
                style="font-size: 16px;"
                tabindex="-1"
                type="text"
                value=""
              />
              <label>
                <select
                  disabled=""
                  multiple=""
                  tabindex="-1"
                  title=""
                >
                  <option />
                </select>
              </label>
            </div>
            <div
              class="w-full flex flex-col"
              data-slot="mainWrapper"
            >
              <button
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-label="Price"
                aria-labelledby="react-aria-:r27: react-aria-:r22:"
                class="relative gap-3 w-full inline-flex flex-row items-center shadow-sm tap-highlight-transparent bg-default-100 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 pointer-events-none outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 min-h-9 h-9 px-3.5 rounded-[28px]"
                data-disabled="true"
                data-slot="trigger"
                data-testid="Price"
                id="react-aria-:r22:"
                type="button"
              >
                <div
                  class="inline-flex h-full w-[calc(100%_-_theme(spacing.6))] min-h-4 items-center gap-1.5 box-border"
                  data-slot="innerWrapper"
                >
                  <span
                    class="w-full text-left rtl:text-right truncate group-data-[has-value=true]:text-default-foreground text-primary-200 font-medium text-sm"
                    data-slot="value"
                    id="react-aria-:r27:"
                  >
                    Price
                  </span>
                </div>
                <svg
                  aria-hidden="true"
                  class="absolute right-3 rtl:left-3 rtl:right-[unset] w-4 h-4 transition-transform duration-150 ease motion-reduce:transition-none data-[open=true]:rotate-180"
                  data-slot="selectorIcon"
                  fill="none"
                  focusable="false"
                  height="1em"
                  role="presentation"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="1em"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="group flex-col pointer-events-none transition-background motion-reduce:transition-none !duration-150 group relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)] rounded-[28px] opacity-100 bg-background-400 text-primary-200 font-medium text-sm w-[102px] lg:block hidden"
            data-filled="true"
            data-slot="base"
          >
            <div
              aria-hidden="true"
              data-a11y-ignore="aria-hidden-focus"
              data-testid="hidden-select-container"
              style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
            >
              <input
                disabled=""
                style="font-size: 16px;"
                tabindex="-1"
                type="text"
                value=""
              />
              <label>
                <select
                  disabled=""
                  multiple=""
                  tabindex="-1"
                  title=""
                >
                  <option />
                </select>
              </label>
            </div>
            <div
              class="w-full flex flex-col"
              data-slot="mainWrapper"
            >
              <button
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-label="Review"
                aria-labelledby="react-aria-:r2f: react-aria-:r2a:"
                class="relative gap-3 w-full inline-flex flex-row items-center shadow-sm tap-highlight-transparent bg-default-100 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 pointer-events-none outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 min-h-9 h-9 px-3.5 rounded-[28px]"
                data-disabled="true"
                data-slot="trigger"
                data-testid="Review"
                id="react-aria-:r2a:"
                type="button"
              >
                <div
                  class="inline-flex h-full w-[calc(100%_-_theme(spacing.6))] min-h-4 items-center gap-1.5 box-border"
                  data-slot="innerWrapper"
                >
                  <span
                    class="w-full text-left rtl:text-right truncate group-data-[has-value=true]:text-default-foreground text-primary-200 font-medium text-sm"
                    data-slot="value"
                    id="react-aria-:r2f:"
                  >
                    Review
                  </span>
                </div>
                <svg
                  aria-hidden="true"
                  class="absolute right-3 rtl:left-3 rtl:right-[unset] w-4 h-4 transition-transform duration-150 ease motion-reduce:transition-none data-[open=true]:rotate-180"
                  data-slot="selectorIcon"
                  fill="none"
                  focusable="false"
                  height="1em"
                  role="presentation"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="1em"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="group flex-col pointer-events-none transition-background motion-reduce:transition-none !duration-150 group relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)] rounded-[28px] opacity-100 bg-background-400 text-primary-200 font-medium text-sm w-[98px] lg:block hidden"
            data-filled="true"
            data-slot="base"
          >
            <div
              aria-hidden="true"
              data-a11y-ignore="aria-hidden-focus"
              data-testid="hidden-select-container"
              style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
            >
              <input
                disabled=""
                style="font-size: 16px;"
                tabindex="-1"
                type="text"
                value=""
              />
              <label>
                <select
                  disabled=""
                  multiple=""
                  tabindex="-1"
                  title=""
                >
                  <option />
                </select>
              </label>
            </div>
            <div
              class="w-full flex flex-col"
              data-slot="mainWrapper"
            >
              <button
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-label="Color"
                aria-labelledby="react-aria-:r2n: react-aria-:r2i:"
                class="relative gap-3 w-full inline-flex flex-row items-center shadow-sm tap-highlight-transparent bg-default-100 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 pointer-events-none outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 min-h-9 h-9 px-3.5 rounded-[28px]"
                data-disabled="true"
                data-slot="trigger"
                data-testid="Color"
                id="react-aria-:r2i:"
                type="button"
              >
                <div
                  class="inline-flex h-full w-[calc(100%_-_theme(spacing.6))] min-h-4 items-center gap-1.5 box-border"
                  data-slot="innerWrapper"
                >
                  <span
                    class="w-full text-left rtl:text-right truncate group-data-[has-value=true]:text-default-foreground text-primary-200 font-medium text-sm"
                    data-slot="value"
                    id="react-aria-:r2n:"
                  >
                    Color
                  </span>
                </div>
                <svg
                  aria-hidden="true"
                  class="absolute right-3 rtl:left-3 rtl:right-[unset] w-4 h-4 transition-transform duration-150 ease motion-reduce:transition-none data-[open=true]:rotate-180"
                  data-slot="selectorIcon"
                  fill="none"
                  focusable="false"
                  height="1em"
                  role="presentation"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="1em"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="group flex-col pointer-events-none transition-background motion-reduce:transition-none !duration-150 group relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)] rounded-[28px] opacity-100 bg-background-400 text-primary-200 font-medium text-sm w-[108px] lg:block hidden"
            data-filled="true"
            data-slot="base"
          >
            <div
              aria-hidden="true"
              data-a11y-ignore="aria-hidden-focus"
              data-testid="hidden-select-container"
              style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
            >
              <input
                disabled=""
                style="font-size: 16px;"
                tabindex="-1"
                type="text"
                value=""
              />
              <label>
                <select
                  disabled=""
                  multiple=""
                  tabindex="-1"
                  title=""
                >
                  <option />
                </select>
              </label>
            </div>
            <div
              class="w-full flex flex-col"
              data-slot="mainWrapper"
            >
              <button
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-label="Material"
                aria-labelledby="react-aria-:r2v: react-aria-:r2q:"
                class="relative gap-3 w-full inline-flex flex-row items-center shadow-sm tap-highlight-transparent bg-default-100 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 pointer-events-none outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 min-h-9 h-9 px-3.5 rounded-[28px]"
                data-disabled="true"
                data-slot="trigger"
                data-testid="Material"
                id="react-aria-:r2q:"
                type="button"
              >
                <div
                  class="inline-flex h-full w-[calc(100%_-_theme(spacing.6))] min-h-4 items-center gap-1.5 box-border"
                  data-slot="innerWrapper"
                >
                  <span
                    class="w-full text-left rtl:text-right truncate group-data-[has-value=true]:text-default-foreground text-primary-200 font-medium text-sm"
                    data-slot="value"
                    id="react-aria-:r2v:"
                  >
                    Material
                  </span>
                </div>
                <svg
                  aria-hidden="true"
                  class="absolute right-3 rtl:left-3 rtl:right-[unset] w-4 h-4 transition-transform duration-150 ease motion-reduce:transition-none data-[open=true]:rotate-180"
                  data-slot="selectorIcon"
                  fill="none"
                  focusable="false"
                  height="1em"
                  role="presentation"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="1em"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="group flex-col pointer-events-none transition-background motion-reduce:transition-none !duration-150 group relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)] rounded-[28px] opacity-100 bg-background-400 text-primary-200 font-medium text-sm w-[88px] lg:block hidden"
            data-filled="true"
            data-slot="base"
          >
            <div
              aria-hidden="true"
              data-a11y-ignore="aria-hidden-focus"
              data-testid="hidden-select-container"
              style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
            >
              <input
                disabled=""
                style="font-size: 16px;"
                tabindex="-1"
                type="text"
                value=""
              />
              <label>
                <select
                  disabled=""
                  multiple=""
                  tabindex="-1"
                  title=""
                >
                  <option />
                </select>
              </label>
            </div>
            <div
              class="w-full flex flex-col"
              data-slot="mainWrapper"
            >
              <button
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-label="Offer"
                aria-labelledby="react-aria-:r37: react-aria-:r32:"
                class="relative gap-3 w-full inline-flex flex-row items-center shadow-sm tap-highlight-transparent bg-default-100 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 pointer-events-none outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 min-h-9 h-9 px-3.5 rounded-[28px]"
                data-disabled="true"
                data-slot="trigger"
                data-testid="Offer"
                id="react-aria-:r32:"
                type="button"
              >
                <div
                  class="inline-flex h-full w-[calc(100%_-_theme(spacing.6))] min-h-4 items-center gap-1.5 box-border"
                  data-slot="innerWrapper"
                >
                  <span
                    class="w-full text-left rtl:text-right truncate group-data-[has-value=true]:text-default-foreground text-primary-200 font-medium text-sm"
                    data-slot="value"
                    id="react-aria-:r37:"
                  >
                    Offer
                  </span>
                </div>
                <svg
                  aria-hidden="true"
                  class="absolute right-3 rtl:left-3 rtl:right-[unset] w-4 h-4 transition-transform duration-150 ease motion-reduce:transition-none data-[open=true]:rotate-180"
                  data-slot="selectorIcon"
                  fill="none"
                  focusable="false"
                  height="1em"
                  role="presentation"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="1em"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="group flex-col pointer-events-none transition-background motion-reduce:transition-none !duration-150 group relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)] rounded-[28px] opacity-100 text-primary-200 font-medium text-sm w-[166px] bg-background-100 border-small border-foreground-300 lg:block hidden"
          data-filled="true"
          data-slot="base"
        >
          <div
            aria-hidden="true"
            data-a11y-ignore="aria-hidden-focus"
            data-testid="hidden-select-container"
            style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
          >
            <input
              disabled=""
              style="font-size: 16px;"
              tabindex="-1"
              type="text"
              value=""
            />
            <label>
              <select
                disabled=""
                multiple=""
                tabindex="-1"
                title=""
              >
                <option />
              </select>
            </label>
          </div>
          <div
            class="w-full flex flex-col"
            data-slot="mainWrapper"
          >
            <button
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-label="Headphone type"
              aria-labelledby="react-aria-:r3f: react-aria-:r3a:"
              class="relative gap-3 w-full inline-flex flex-row items-center shadow-sm tap-highlight-transparent bg-default-100 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 pointer-events-none outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 min-h-9 h-9 px-3.5 rounded-[28px]"
              data-disabled="true"
              data-slot="trigger"
              data-testid="Headphone type"
              id="react-aria-:r3a:"
              type="button"
            >
              <div
                class="inline-flex h-full w-[calc(100%_-_theme(spacing.6))] min-h-4 items-center gap-1.5 box-border"
                data-slot="innerWrapper"
              >
                <span
                  class="w-full text-left rtl:text-right truncate group-data-[has-value=true]:text-default-foreground text-primary-200 font-medium text-sm"
                  data-slot="value"
                  id="react-aria-:r3f:"
                >
                  Headphone type
                </span>
              </div>
              <svg
                aria-hidden="true"
                class="absolute right-3 rtl:left-3 rtl:right-[unset] w-4 h-4 transition-transform duration-150 ease motion-reduce:transition-none data-[open=true]:rotate-180"
                data-slot="selectorIcon"
                fill="none"
                focusable="false"
                height="1em"
                role="presentation"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="1em"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="card-list grid grid-cols-2 gap-x-[20px] gap-y-[30px] mb-[110px] w-fit mx-auto 2xl:grid-cols-4 lg:grid-cols-3"
      />
    </div>
  </div>
</div>
`;

exports[`ProductList should match snapshot value error 1`] = `
<div>
  <div
    class="container w-full max-w-full xl:py-16 lg:py-12 py-8 "
  >
    <div
      class="w-full"
    >
      <div
        class="flex sm:flex-row flex-col gap-4 items-center lg:justify-between justify-end xl:mb-12 mb-6"
      >
        <div
          class="flex gap-3"
        />
      </div>
      <div
        class="flex flex-col min-h-[320px] h-[320px] gap-4 items-center justify-center"
      >
        <div
          class="flex flex-col h-full gap-4 items-center justify-center"
        >
          <p
            class="  no-underline text-ellipsis whitespace-nowrap overflow-hidden text-primary-300 font-bold md:text-5xl text-3xl md:leading-10 leading-[28px]"
          >
            Empty product
          </p>
          <a
            class="mt-3"
            href="/"
          >
            <button
              aria-label="Button"
              class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover gap-2.5 font-semibold bg-primary-300 text-background-100 sm:py-4 sm:min-w-[136px] sm:h-12 sm:text-base sm:rounded-[50px] py-2 min-w-20 h-9 text-sm rounded-[30px]"
              type="button"
            >
              Go Home
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
